src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -218,16 +212,16 @@ public boolean equals(Object obj) {
                Objects.equals(queryName, other.queryName);
     }
 
-    /** Read the given parameters. */
     @Override
-    public void readFrom(StreamInput in) throws IOException {
-        fieldName = in.readString();
-        value = in.readGenericValue();
-        boost = in.readFloat();
-        queryName = in.readOptionalString();
+    public QB readFrom(StreamInput in) throws IOException {
+        QB emptyBuilder = createBuilder(in.readString(), in.readGenericValue());
+        emptyBuilder.boost(in.readFloat());
```

Comment:
I don't think setting fields directly is bad practice, we could do that if we prefer it actually.